test/src/transport/ClientRemotingProcessorTest.cpp

```diff
@@ -0,0 +1,235 @@
+    ClientRemotingProcessor clientRemotingProcessor(factory);
+
+    string addr = "127.0.0.1:9876";
+    RemotingCommand *command = new RemotingCommand();
+    RemotingCommand *pResponse = new RemotingCommand(13);
+
+    pResponse->setCode(MQRequestCode::RESET_CONSUMER_CLIENT_OFFSET);
+    command->setCode(MQRequestCode::RESET_CONSUMER_CLIENT_OFFSET);
+    EXPECT_TRUE(clientRemotingProcessor.processRequest(addr, command) == nullptr);
+    EXPECT_EQ(nullptr, clientRemotingProcessor.processRequest(addr, command));
+
+    NotifyConsumerIdsChangedRequestHeader *header = new NotifyConsumerIdsChangedRequestHeader();
+    header->setGroup("testGroup");
+    RemotingCommand *twoCommand = new RemotingCommand(MQRequestCode::NOTIFY_CONSUMER_IDS_CHANGED, header);
+
+    EXPECT_EQ(NULL, clientRemotingProcessor.processRequest(addr, twoCommand));
+
+    command->setCode(MQRequestCode::GET_CONSUMER_RUNNING_INFO);
+    // EXPECT_EQ(NULL , clientRemotingProcessor.processRequest(addr, command));
+
+    command->setCode(MQRequestCode::CHECK_TRANSACTION_STATE);
+    EXPECT_TRUE(clientRemotingProcessor.processRequest(addr, command) == nullptr);
+
+    command->setCode(MQRequestCode::GET_CONSUMER_STATUS_FROM_CLIENT);
+    EXPECT_TRUE(clientRemotingProcessor.processRequest(addr, command) == nullptr);
+
+    command->setCode(MQRequestCode::CONSUME_MESSAGE_DIRECTLY);
+    EXPECT_TRUE(clientRemotingProcessor.processRequest(addr, command) == nullptr);
+
+    command->setCode(1);
+    EXPECT_TRUE(clientRemotingProcessor.processRequest(addr, command) == nullptr);
+
+    delete command;
+    delete pResponse;
+}
+
+TEST(clientRemotingProcessor, resetOffset) {
+    MockMQClientFactory *factory = new MockMQClientFactory("testClientId", 4, 3000, 4000, "a");
+    Mock::AllowLeak(factory);
+    ClientRemotingProcessor clientRemotingProcessor(factory);
+    Value root;
+    Value messageQueues;
+    Value messageQueue;
+    messageQueue["brokerName"] = "testBroker";
+    messageQueue["queueId"] = 4;
+    messageQueue["topic"] = "testTopic";
+    messageQueue["offset"] = 1024;
+
+    messageQueues.append(messageQueue);
+    root["offsetTable"] = messageQueues;
+
+    FastWriter wrtier;
+    string strData = wrtier.write(root);
+
+    ResetOffsetRequestHeader *header = new ResetOffsetRequestHeader();
+    RemotingCommand *request = new RemotingCommand(13, header);
+
+    EXPECT_CALL(*factory, resetOffset(_, _, _)).Times(1);
+    clientRemotingProcessor.resetOffset(request);
+
+    request->SetBody(strData.c_str(), strData.size() - 2);
+    clientRemotingProcessor.resetOffset(request);
+
+    request->SetBody(strData.c_str(), strData.size());
+    clientRemotingProcessor.resetOffset(request);
+
+    delete header;
+    delete request;
+}
+
+TEST(clientRemotingProcessorS, getConsumerRunningInfo) {
+    MockMQClientFactory *factory = new MockMQClientFactory("testClientId", 4, 3000, 4000, "a");
```

Comment:
factory and info not deleted,  please check the other TEST for not free the object, or use shared_ptr.   also i think you can use local object, no need with new

Reply:
Mock::AllowLeak(factory);   Be equal to deleted

Reply:
but there are many point without Mock::AllowLeak